data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -73,23 +73,65 @@ _ "This attack uses 6 attacks"#enddef
     [/event]
 #enddef
 
-#define WEAPON_SPECIAL_SONG_VERSE ACTIVE_ON X DISTINGUISHER
+#define STR_VERSE_RESTRICT_1
+_ "This attack is the first verse of the Song of Sun Ascension, which progresses by one verse each turn this unit uses a song verse attack. When the last verse is sung or the song sequence is broken due to not attacking for a turn, the song starts again from the first verse."#enddef
+
+#define STR_VERSE_RESTRICT_2
+_ "This attack the second verse of the Song of Sun Ascension, which progresses by one verse each turn this unit uses a song verse attack. When the last verse is sung or the song sequence is broken due to not attacking for a turn, the song starts again from the first verse."#enddef
+
+#define STR_VERSE_RESTRICT_3
+_ "This attack the third verse of the Song of Sun Ascension, which progresses by one verse each turn this unit uses a song verse attack. When the last verse is sung or the song sequence is broken due to not attacking for a turn, the song starts again from the first verse."#enddef
+
+#define STR_VERSE_NAME_1
+_ "first verse"#enddef
+
+#define STR_VERSE_NAME_2
+_ "second verse"#enddef
+
+#define STR_VERSE_NAME_3
+_ "third verse"#enddef
+
+# param MAX is the total number of verses the unit has, needed for id
+
+#define WEAPON_SPECIAL_SONG_VERSE_BASE MAX
+    [disable]
+        id = verse_{MAX}
+        [filter_self]
+            formula = "self.wml_vars.current_verse > 1"
+        [/filter_self]
+        active_on = offense
+        name = {STR_VERSE_NAME_1}
+        description = {STR_VERSE_RESTRICT_1}
+    [/disable]
+#enddef
+
+# param MAX is the this attack should be active on
+# param MAX is the total number of verses the unit has, needed for id
+# example: {WEAPON_SPECIAL_SONG_VERSE 2 3} means this attack should be active on second verse out of three
+# don't set MAX to 1, use WEAPON_SPECIAL_SONG_VERSE_BASE instead to prevent an obscure error related to debug/script unit placement
+
+#define WEAPON_SPECIAL_SONG_VERSE VERSE MAX
     [disable]
-        id = verse_{DISTINGUISHER}
+        id = verse_{MAX}
         [filter_self]
-            formula = "self.wml_vars.consecutive_attacks != {X}"
+            formula = "self.wml_vars.current_verse != {VERSE}"
         [/filter_self]
-        {ACTIVE_ON}
+        name = {STR_VERSE_NAME_{VERSE}}
+        description = {STR_VERSE_RESTRICT_{VERSE}}
     [/disable]
 #enddef
 
-#define SONG_VERSE_EVENTS X DISTINGUISHER
+# VERSE is the number of verses the unit has
+# this macro must be called for each unique VERSE
```

Comment:
"The number of verses the unit has" sounds more like MAX than VERSE, but from this documentation I'm unsure which it should be.

Reply:
VERSE, it is used in resetting back to 1

Reply:
SONG_VERSE_EVENTS' documentation still talks about VERSE, but it now takes only MAX. Please update the docs to explain what it does, not just to update the variable name.